mono/metadata/profiler.c

```diff
@@ -1147,7 +1147,7 @@ load_embedded_profiler (const char *desc, const char *name)
 	MonoDl *pmodule = NULL;
 	gboolean result;
 
-	pmodule = mono_dl_open (NULL, MONO_DL_LAZY, &err);
+	pmodule = mono_dl_open (NULL, 0, &err);
```

Comment:
The fact that passing in 0 instead of MONO_DL_LAZY results in eager loading, and the fact that eager loading is required because the profiler contains functions that run inside signal handlers, are both very nonobvious. Please add a comment to explain this decision and the reasoning for it.